### PR TITLE
[MRG+1] Add results obtained for Branin and Hart6

### PR DESCRIPTION
**Pull request by MechCoder**
An attempt to reproduce the extensive benchmarks described in http://www.automl.org/papers/13-BayesOpt_EmpiricalFoundation.pdf

TLDR; We are worse than Spearmint but better than others (for 2 benchmark functions)

### CODE REVIEW

**Comment by MechCoder** [comment]
Note that this was before Spearmint implemented the beta transform and other improvements.

**Comment by MechCoder** [comment]
Scripts to reproduce: https://github.com/MechCoder/Notebooks/blob/master/Function%20benchmarks.ipynb

**Comment by betatim** [comment]
I'd store the results in a subdirectory.

Can we turn the notebook into a script that can live next to the results? Otherwise I fear we will "forget" how the numbers were produced or how to reproduce them or add more benchmarks.

**Comment by MechCoder** [comment]
Sure, but I would prefer https://github.com/scikit-optimize/scikit-optimize/pull/215 to be merged before this. It exposes the time taken for each iteration to the user, so with that we can keep a track of the time as well and log that as well. WDYT?

**Comment by MechCoder** [comment]
y@betatim Addressed your comments. Let me know if there is anything else.

**Comment by MechCoder** [comment]
How about the other minimize functions. Do we still bench with `n_random_starts=1` despite knowing that this will default to random search until we cover the edges of the search space?

**Comment by betatim** [comment]
> Do we still bench with n_random_starts=1 despite knowing that this will default to random search until we cover the edges of the search space?

why would we use `n_random_starts=1`? I'd use our default values as much as possible.

File: benchmarks/RESULTS.md [hunk 1]
```diff
@@ -0,0 +1,10 @@
+These are the results got by optimizing the particular set of benchmark functions provided in Table 2 of
+(http://www.automl.org/papers/13-BayesOpt_EmpiricalFoundation.pdf) with our defualt values. All optimizers are run 10 times with `n_calls=200`.
+
+## gp_minimize
+
+
+|Blackbox Function| Valid loss | Best loss |
```
Comment (betatim) [inline benchmarks/RESULTS.md]: Valid loss? Isn't "average minimum" better? As well as "best minimum"?

Comment (MechCoder) [inline benchmarks/RESULTS.md]: I just followed the convention in the paper.

**Comment by MechCoder** [comment]
Running benchmarks for `forest` and `gbrt`

**Comment by MechCoder** [comment]
One quick look that I haven't done anything silly with the benchmark scripts and it should be good for merge. (Surprised that `forest_minimize` performs better on hart6)

File: benchmarks/RESULTS.md (hunk 1 above)
Comment (betatim) [inline benchmarks/RESULTS.md]: Ah ok,  ¯\_(ツ)_/¯

**Comment by betatim** [comment]
I'll take a look tomorrow morning.

**Comment by MechCoder** [comment]
also opened an issue here (https://github.com/automl/HPOlib/issues/114)

**Comment by betatim** [comment]
LGTM. If you wanted to you could rename `RESULTS.md` to `README.md` so it gets rendered automagically when you navigate to the directory. Merge with or without that change 😃

**Pull request merged**